iroh-gossip/src/net.rs

```diff
@@ -101,7 +106,10 @@ impl Gossip {
             subscribers_topic: Default::default(),
         };
         let actor_handle = tokio::spawn(async move {
-            if let Err(err) = actor.run().await {
+            let fut = actor.run();
+            #[cfg(feature = "log-self")]
+            let fut = fut.instrument(trace_span!("gossip", %me));
```

Comment:
since this is a spawned future I'd strongly recommend to use an `info_span`.  And also to not make it depend on the feature.  Every task should have it's own `info_span` attached as a general rule.

The span should also cover the bit of error handing in the lines below that are still inside this spawned task.

Reply:
ok just to be clear I don't understand what this changes, not sure what the effect of this is

Reply:
See also https://discord.com/channels/949724860232392765/950683937661935667/1156506832567808060 where @Frando is also asked me this question and I (hopefully!) gave a similar answer.